File: backend/app/models/code_interpreter.py
from pydantic import BaseModel
from typing import Optional

class CodeInterpreterState(BaseModel):
    id: str
    ticket_id: str
    code: str
    description: Optional[str] = ""
    status: str = "pending"  # pending, running, completed, error, approved
    result: Optional[str] = None
    widget_url: Optional[str] = None
    created_at: int
    updated_at: int

class CodeInterpreterCreateRequest(BaseModel):
    code: str
    description: Optional[str] = ""

class CodeInterpreterUpdateRequest(BaseModel):
    status: Optional[str] = None
    result: Optional[str] = None

class CodeInterpreterResponse(BaseModel):
    id: str
    ticket_id: str
    status: str
    widget_url: Optional[str] = None
    result: Optional[str] = None

File: frontend/src/types/code_interpreter.ts
export interface CodeInterpreterState {
  id: string;
  session_id: string;
  ticket_id: string;
  code: string;
  description?: string;
  status: 'pending' | 'running' | 'completed' | 'error';
  result?: string;
  widget_url?: string;
  created_at: number;
  updated_at: number;
}

export interface CodeInterpreterCreateRequest {
  session_id: string;
  code: string;
  description?: string;
}

export interface CodeInterpreterUpdateRequest {
  status?: 'pending' | 'running' | 'completed' | 'error';
  result?: string;
}

export interface CodeInterpreterHookReturn {
  states: CodeInterpreterState[];
  loading: boolean;
  error: string | null;
  createState: (sessionId: string, code: string, description?: string) => Promise<void>;
  updateState: (id: string, updates: Partial<CodeInterpreterState>) => Promise<void>;
  getState: (id: string) => Promise<void>;
  fetchStates: () => Promise<void>;
}
